pkg/util/fxutil/logging/logging.go

```diff
@@ -16,13 +17,16 @@ import (
 // DefaultFxLoggingOption creates an fx.Option to configure the Fx logger, either to do nothing
 // (the default) or to log to the console (when TRACE_FX is set at `1`).
 func DefaultFxLoggingOption() fx.Option {
+	starttime := time.Now()
 	return fx.Options(
 		fx.WithLogger(
 			func() fxevent.Logger {
 				if os.Getenv("TRACE_FX") == "1" {
-					return &fxevent.ConsoleLogger{W: os.Stderr}
+					// We log to stderr to avoid polluting the log file with Fx events
+					return withFxTracer(&fxevent.ConsoleLogger{W: os.Stderr}, starttime, os.Stderr)
 				}
-				return fxevent.NopLogger
+				// We log to stderr to avoid polluting the log file with Fx events
+				return withFxTracer(fxevent.NopLogger, starttime, os.Stderr)
```

Comment:
Tiny nit: this would avoid some duplication, and I think makes the logic clearer
```go
var logger fxevent.Logger = fxevent.NopLogger
if os.Getenv("TRACE_FX") == "1" {
	logger = &fxevent.ConsoleLogger{W: os.Stderr}
}
return withFxTracer(logger, starttime, os.Stderr)
```